modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -1287,9 +1321,37 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
 
         if (openApiNullable) {
             if (Boolean.FALSE.equals(property.required) && Boolean.TRUE.equals(property.isNullable)) {
+                // Only add import when needed
                 model.imports.add("JsonNullable");
                 model.getVendorExtensions().put("x-jackson-optional-nullable-helpers", true);
             }
+            if (SpringCodegen.class.equals(this.getClass()) && Boolean.FALSE.equals(property.required) && Boolean.TRUE.equals(property.isNullable)) {
+                // Wrap dataType and defaults
+                property.isWrapped = true;
+                property.wrapperType = "JsonNullable";
+                property.wrapperFunc = "JsonNullable.of";
+                if (property.defaultValue != null) {
+                    property.defaultValue = "JsonNullable.of(" + property.defaultValue + ")";
+                } else {
+                    property.defaultValue = "JsonNullable.undefined()";
+                }
+            }
+        }
+        if (useOptional) {
```

Comment:
I think if optionals are enabled, it would be nice for non-nullable non-required maps/arrays to default to empty map/array instead of `null` since an empty list/map has the same meanining as a `null` one in that case.